content/en/docs/handling-errors.md

```diff
@@ -2,76 +2,157 @@
-interface type and the `awserr.Error` interface in the
-``aws/awserr`` package. You can use the `Error()` method to get a formatted string of
-the SDK error message without any special handling.
+The {{% alias sdk-go %}} returns errors that satisfy the Go `error` interface type  You can use the `Error()` method to
+get a formatted string of the SDK error message without any special handling. Errors returned by the SDK may implement
+an `Unwrap` method. The `Unwrap` method is used by the SDK to provide additional contextual information to errors, while
+providing access to the underlying error or chain of errors. The `Unwrap` method should be used with the
+[errors.As](https://golang.org/pkg/errors#As) and [errors.Is](https://golang.org/pkg/errors#Is) functions which can
+handle unpacking wrapped error chains.
+
+It is important that your application check whether an error occurred after invoking a function or method that
+can return an `error` interface type. The most basic form of error handling looks similar to the following example:
+
+```go
+if err != nil {
+	// Handle error
+	return
+}
+```
+
+## Logging Errors
+
+The simplest form of error handling is traditionally to log or print the error message before returning or exiting from
+the application. For example:
+
+```go
+import "log"
+
+// ...
+
+if err != nil {
+	log.Printf("error: %s", err.Error())
+	return
+}
+```
+
+## Service Client Errors
+
+The SDK wraps All errors returned by service clients with the
+[smithy.OperationError]({{% apiref smithy="#OperationError" %}}) error type. `OperationError` provides contextual
+information about the service name and operation that is associated with an underlying error. This information can be
+useful for applications that perform batches of operations to one or more services, with a centralized error handling
+mechanism. Your application can use `errors.As` to access this `OperationError` metadata.
+
+For example:
+
+```go
+import "log"
+import "github.com/awslabs/smithy-go"
+
+// ...
+
+if err != nil {
+	var oe *smithy.OperationError
+	if errors.As(err, &oe) {
+		log.Printf("failed to call service: %s, operation: %s, error: %v", oe.Service(), oe.Operation(), oe.Unwrap())
+    }
+    return
+}
+```
+
+### API Error Responses
+
+Service operations can return modeled error types to indicate specific errors. These modeled types can be used with
+`errors.As` to unwrap and determine if the operation failure was due to a specific error. For example
+{{% alias service=S3 %}} `CreateBucket` can return a
+[BucketAlreadyExists]({{< apiref "service/s3/types#BucketAlreadyExists" >}}) error if a bucket of the same name
+already exists.
+
+For example, to check if an error was a `BucketAlreadyExists` error:
 
 ```go
+import "log"
+import "github.com/aws/aws-sdk-go-v2/service/s3/types"
+
+// ...
+
 if err != nil {
-  if awsErr, ok := err.(awserr.Error); ok {
-      // process SDK error
-  }
+	var bne *types.BucketAlreadyExists
```

Comment:
`bne` needs to be a pointer?

Reply:
Yes as it implements error as a pointer receiver.